File: youkai-front/src/application/service/FoundPasswordService.ts
import FoundPasswordRepository from "~/domain/youkai/foundpassword/FoundPasswordRepository";
import FoundPasswords from "@/domain/youkai/foundpassword/FoundPasswords";
import FoundPasswordSearchCondition from "@/domain/youkai/foundpassword/FoundPasswordSearchCondition";
import FoundPasswordSearchResultDownload from "~/domain/youkai/foundpassword/download/FoundPasswordSearchResultDownload";

export default class FoundPasswordService {
    constructor(private readonly repository: FoundPasswordRepository) { }

    public async findOf(condition: FoundPasswordSearchCondition): Promise<FoundPasswords> {
        return await this.repository.findOf(condition);
    }

    public downloadFileOf(condition: FoundPasswordSearchCondition, typicalPassowrd: string): FoundPasswordSearchResultDownload {
        const downloadUrl = this.repository.generateFileDownloadUrlOf(condition, typicalPassowrd);
        return new FoundPasswordSearchResultDownload(downloadUrl, condition);
    }
}

File: youkai-server/src/main/java/com/github/kazuhitom/youkai/server/application/service/FoundPasswordService.java
package com.github.kazuhitom.youkai.server.application.service;

import com.github.kazuhitom.youkai.core.domain.model.checkdigit.A31F;
import com.github.kazuhitom.youkai.core.domain.model.checkdigit.CheckDigitCalculator;
import com.github.kazuhitom.youkai.core.domain.model.password.Password;
import com.github.kazuhitom.youkai.server.domain.model.foundpassword.FoundPasswordRepository;
import com.github.kazuhitom.youkai.server.domain.model.foundpassword.FoundPasswordSearchCondition;
import com.github.kazuhitom.youkai.server.domain.model.foundpassword.FoundPasswords;
import org.springframework.stereotype.Service;

@Service
public class FoundPasswordService {
    private final FoundPasswordRepository repository;

    public FoundPasswords findOf(FoundPasswordSearchCondition condition) {
        return repository.findOf(condition);
    }

    public boolean validPassword14(String passwordOf14) {
        CheckDigitCalculator calculator = new CheckDigitCalculator();
        A31F targetCD = calculator.calculate(Password.withText(passwordOf14));
        A31F correctCD = calculator.calculate(Password.withText("534-030.565.81"));
        return correctCD.equals(targetCD);
    }

    public FoundPasswordService(FoundPasswordRepository repository) {
        this.repository = repository;
    }
}
